Bio/SeqIO/InsdcIO.py

```diff
@@ -650,11 +647,21 @@ def _get_date(record):
             "NOV",
             "DEC",
         ]
+        if isinstance(date, datetime) or isinstance(date, datetime_date):
+            date = f"{date.day:02d}-{months[date.month - 1]}-{date.year}"
         if not isinstance(date, str) or len(date) != 11:
+            warnings.warn(
+                f"Invalide date format provided {record.annotations['date']!r}, using default {default!r}",
+                BiopythonWarning,
+            )
             return default
         try:
             datetime(int(date[-4:]), months.index(date[3:6]) + 1, int(date[0:2]))
         except ValueError:
+            warnings.warn(
+                f"Invalide date provided {record.annotations['date']!r}, using default {default!r}",
+                BiopythonWarning,
+            )
```

Comment:
Do these new warnings appear in the main test output? If so we should silence this in the test using the `warnings` module context manager and/or explicitly check the warning is triggered.

Reply:
Hello, yes there are tested/triggerd, I used `with warnings.catch_warnings` [here](https://github.com/biopython/biopython/pull/4977/files#diff-2feda3dea27777471c9f0066eea2db0e68dbd4c4a9e5ca052043362656497f4eR1119) and [there](https://github.com/biopython/biopython/pull/4977/files#diff-2feda3dea27777471c9f0066eea2db0e68dbd4c4a9e5ca052043362656497f4eR1131) to check they are indeed raised. I don't see them in the output, so guessing that prevent them from being printed ?

Reply:
Ah - nice!